crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1006,6 +1007,30 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                     pyupgrade::rules::unicode_kind_prefix(checker, string_literal);
                 }
             }
+            for literal in value.elements().filter_map(|element| element.as_literal()) {
```

Comment:
This is a good point and yes it will be a false negative now. It's not just `f"/tmp" f"/bad"` but also `"/tmp" f"/bad"` (string and f-string)

We can introduce a method which iterates over the concatenated literal parts of an f-string. For example,

```python
"foo" f"bar {x} baz" "end"
```

The above code would return two strings: `"foobar "` and `" bazend"`. Here, I don't see a way to avoid string allocation here.

Reply:
I'm moving ahead with this limitation for now but we can revisit if it proves to be a problem. My hunch is this shouldn't be but you never know 🤷‍♂️